app/src/main/java/org/simple/clinic/registration/register/RegistrationLoadingScreenController.kt

```diff
@@ -30,6 +30,7 @@ class RegistrationLoadingScreenController @Inject constructor(
 
     return Observable.merge(creates, retryClicks)
         .flatMapSingle { userSession.register() }
+        .doOnNext { if (it is Success) userSession.clearOngoingRegistrationEntry().subscribe() }
```

Comment:
This is a dangling `subscribe` call and is a code smell. What we may need here is a `flatMap` or fork the stream and process them separately.